src/main/java/org/apache/pulsar/manager/service/impl/GithubLoginServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.pulsar.manager.service.ThirdLoginService;
+import org.apache.pulsar.manager.utils.HttpUtil;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+
+/**
+ * Github login for get user information.
+ */
+@Slf4j
+@Service
+public class GithubLoginServiceImpl implements ThirdLoginService {
+
+    @Value("${github.client.id}")
+    private String githubClientId;
+
+    @Value("${github.client.secret}")
+    private String githubClientSecret;
+
+    @Value("${github.oauth.host}")
+    private String githubAuthHost;
+
+    @Value("${github.user.info}")
+    private String githubUserInfo;
+
+    /**
+     * Get user access token from github.
+     * @param parameters For get code to github.
+     *          GitHub redirects back to local site with a temporary code in a code parameter as well as the state
+     *          you provided in the previous step in a state parameter.The temporary code will expire after 10 minutes.
+     * @return access_token of string type
+     */
+    public String getAuthToken(Map<String, String> parameters) {
+        Map<String, String> header = Maps.newHashMap();
+        header.put("Content-Type", "application/json");
+        header.put("Accept", "application/json");
+        Map<String, Object> body = Maps.newHashMap();
+        body.put("client_id", githubClientId);
+        body.put("client_secret", githubClientSecret);
+        if (!parameters.containsKey("code")) {
+            log.error("Parameter does not contain code field, which is illegal.");
+            return null;
+        }
+        body.put("code", parameters.get("code"));
+        Gson gson = new Gson();
+        try {
+            // result example: access_token=e72e16c7e42f292c6912e7710c838347ae178b4a&token_type=bearer
+            String result = HttpUtil.doPost(githubAuthHost, header, gson.toJson(body));
+            GithubAuthEntity githubAuthEntity = gson.fromJson(result, GithubAuthEntity.class);
+            log.info("Success get access token from github");
+            return githubAuthEntity.getAccessToken();
+        } catch (UnsupportedEncodingException e) {
+            log.error("Failed get access token from github, error stack: {}", e.getCause());
+            return null;
+        }
+    }
+
+    /**
+     * Get user information from github by access token.
+     * @param authenticationMap Authentication mark requesting user information.
+     * @return UserInfoEntity
+     */
+    public UserInfoEntity getUserInfo(Map<String, String> authenticationMap) {
+        Map<String, String> header = Maps.newHashMap();
+        header.put("Content-Type", "application/json");
+        if (!authenticationMap.containsKey("access_token")) {
+            log.error("The authenticationMap does not contain access_token field, which is illegal.");
+            return null;
```

Comment:
did we handle the returned `null` value?

Reply:
Yes, an empty exception was handled during the call https://github.com/apache/pulsar-manager/pull/222/files#diff-8cafd3d67af2d98edba072818ff3dc62R74.